ui/app/src/components/Header/SearchBar/SearchBar.tsx

```diff
@@ -47,6 +68,19 @@ function SearchDashboardList(props: { query: string; onClick: () => void }) {
   });
 }
 
+function SearchDatasourceList(props: ResourceListProps) {
+  const datasourceQueryResult = useDatasourceList(undefined);
+  return SearchList({
+    list: datasourceQueryResult.data ?? [],
+    query: props.query,
+    onClick: props.onClick,
+    icon: DatabaseIcon,
+    chip: true,
+    buildRouting: (resource) =>
+      `${ProjectRoute}/${isProjectMetadata(resource.metadata) ? resource.metadata.project : ''}/datasources`,
+  });
+}
+
 function useHandleShortCut(handleOpen: () => void) {
   // handle what happens on key press
   const handleKeyPress = useCallback(
```

Comment:
Line 90, it's missing the `event.preventDefault()` in order to not propagate to browser